tests/connect/test_join.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from pyspark.sql.functions import col
+
+
+def test_join(spark_session):
+    # Create two DataFrames with overlapping IDs
+    df1 = spark_session.range(5)
+    df2 = spark_session.range(3, 7)
+
+    # Perform inner join on 'id' column
+    joined_df = df1.join(df2, "id", "inner")
+
+    # Verify join results
+    # assert joined_df.count() == 2, "Inner join should only return matching rows"
+    
+    # Convert to pandas to verify exact results
+    joined_pandas = joined_df.toPandas()
+    assert set(joined_pandas["id"].tolist()) == {3, 4}, "Inner join should only contain IDs 3 and 4"
+
+    # Test left outer join
+    left_joined_pandas = df1.join(df2, "id", "left").toPandas()
+    assert set(left_joined_pandas["id"].tolist()) == {0, 1, 2, 3, 4}, "Left join should keep all rows from left DataFrame"
+
+    # Test right outer join 
+    right_joined_pandas = df1.join(df2, "id", "right").toPandas()
+    assert set(right_joined_pandas["id"].tolist()) == {3, 4, 5, 6}, "Right join should keep all rows from right DataFrame"
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +6 to +28]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->we're going to need a lot more tests for joins. Check out the other join tests, both for dataframe and sql. <!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3354?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->